analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.analyzer.managers
+
+import com.here.ort.analyzer.AbstractPackageManagerFactory
+import com.here.ort.analyzer.PackageManager
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.model.*
+import com.here.ort.model.config.AnalyzerConfiguration
+import com.here.ort.model.config.RepositoryConfiguration
+import com.here.ort.utils.CommandLineTool
+
+import java.io.File
+import java.io.IOException
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.module.kotlin.readValues
+
+
+/**
+ * The [Go Modules](https://github.com/golang/go/wiki/Modules) package manager for Go.
+ */
+class GoMod(
+    name: String,
+    analysisRoot: File,
+    analyzerConfig: AnalyzerConfiguration,
+    repoConfig: RepositoryConfiguration
+) : PackageManager(name, analysisRoot, analyzerConfig, repoConfig), CommandLineTool {
+    class Factory : AbstractPackageManagerFactory<GoMod>("GoMod") {
+        override val globsForDefinitionFiles = listOf("go.mod")
+
+        override fun create(
+            analysisRoot: File,
+            analyzerConfig: AnalyzerConfiguration,
+            repoConfig: RepositoryConfiguration
+        ) = GoMod(managerName, analysisRoot, analyzerConfig, repoConfig)
+    }
+
+    override fun command(workingDir: File?): String {
+        return "go"
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    data class ModuleDependency(
+        @JsonProperty("Path") val name: String,
+        @JsonProperty("Dir") val dir: String,
+        @JsonProperty("Version") val version: String?,
+        @JsonProperty("Main") val isMain: Boolean = false
+    )
+
+    override fun resolveDependencies(definitionFile: File): ProjectAnalyzerResult? {
+        val projectDir = definitionFile.parentFile
+
+        if (!analyzerConfig.allowDynamicVersions &&
+            !projectDir.resolve("go.sum").exists()
+        ) {
+            throw IOException("No go.sum file found in '$projectDir'.")
+        }
+
+        installDependencies(projectDir)
```

Comment:
What other package managers do is
- move the directory elsewhere e.g. analog to 'vendor'
- install deps into the new empty vendor dir
- analyze the content
- clean up, recover previous directory state

Would it make to also use `DirectoryStash.stashDirectories`

Reply:
It is beeing used now.